File: SeLeAs/Python/uniqueOccurances.py
from collections import Counter

class Solution:
    def uniqueOccurrences(self, arr):
        # Count the frequency of each element in arr
        count_map = Counter(arr)
        
        # Get the list of frequencies of the elements
        frequencies = list(count_map.values())
        
        # Check if all frequencies are unique
        return len(frequencies) == len(set(frequencies))

File: SeLeAs/Java/uniqueOccurances.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        // Count the frequency of each element in arr
        Map<Integer, Integer> countMap = new HashMap<>();
        for (int num : arr) {
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }
        
        // Create a set to store the frequencies and check if they are unique
        Set<Integer> frequencies = new HashSet<>(countMap.values());
        
        // If the size of the set is equal to the number of unique frequencies, return true
        return countMap.size() == frequencies.size();
    }
}
